packages/ui/src/components/va-select/components/VaSelectContent/VaSelectContent.vue

```diff
@@ -74,16 +118,84 @@ export default defineComponent({
+    focused: { type: Boolean, default: false },
+    dropdownContentShown: { type: Boolean, default: false },
+    multiple: { type: Boolean, default: false },
+    disabled: { type: Boolean, default: false },
+    getText: { type: Function as PropType<(option: SelectOption) => string>, required: true },
   },
 
-  emits: ['toggle-hidden'],
+  emits: ['toggle-hidden', 'autocomplete-input', 'focus-prev', 'focus-next', 'select-option', 'delete-last-selected'],
 
   setup (props, { emit }) {
+    const autocompleteInput = ref<HTMLInputElement>()
+
     const isPlaceholder = computed(() => props.placeholder && !props.valueString)
 
     const toggleHiddenOptionsState = () => emit('toggle-hidden')
 
-    return { isPlaceholder, toggleHiddenOptionsState }
+    const { dropdownContentShown, valueString, value, focused } = toRefs(props)
+
+    const autocompleteInputValue = ref('')
+    const convertToString = (v: number | string | undefined) => v ? String(v) : ''
+    const getLastOptionText = (v: SelectOption[] | undefined) => {
+      return v && Array.isArray(v) && v.length ? props.getText(v.at(-1)!) : ''
+    }
+    props.autocomplete && watch([autocompleteInputValue, valueString, dropdownContentShown, value],
+      (
+        [newAutocompleteValue, newValueString, newDropdownShownState, newValueArray],
+        [oldAutocompleteValue, oldValueString, oldDropdownShownState, oldValueArray],
+      ) => {
+        // just opened, nothing to handle
+        if (newDropdownShownState && !oldDropdownShownState) { return }
+
+        // just cleared after blur, nothing to handle
+        if (autocompleteInputValue.value === '' && !newDropdownShownState) { return }
+
+        let newValueStringConverted = ''
+        let oldValueStringConverted = ''
+
+        // taking string value or last options array text value depending on multiple state
+        if (props.multiple) {
+          newValueStringConverted = getLastOptionText(newValueArray)
+          oldValueStringConverted = getLastOptionText(oldValueArray)
+        } else {
+          newValueStringConverted = convertToString(newValueString)
+          oldValueStringConverted = convertToString(oldValueString)
+        }
+
+        if (newValueStringConverted !== oldValueStringConverted) {
+          // selected value change handling, not for multiple state
+          if (!props.multiple) { autocompleteInputValue.value = newValueStringConverted }
+        } else if (newAutocompleteValue !== oldAutocompleteValue && autocompleteInputValue.value !== newValueStringConverted) {
+          // autocomplete input handling
+          emit('autocomplete-input', newAutocompleteValue)
+        }
+
+        if (!newDropdownShownState && oldDropdownShownState && newValueStringConverted !== autocompleteInputValue.value) {
+          // handling input wrapper blur, not for multiple state
+          autocompleteInputValue.value = props.multiple ? '' : newValueStringConverted
+        }
+      })
```

Comment:
```suggestion
    watch(value, (newVal, oldVal) => {
      if (props.multiple) {
        autocompleteInputValue.value = ''
        return
      }

      if (typeof newVal === 'undefined') {
        autocompleteInputValue.value = ''
        return
      }

      autocompleteInputValue.value = convertToString(newVal as string)
    })

    watch(autocompleteInputValue, (newVal, oldVal) => {
      if (props.autocomplete) {
        emit('autocomplete-input', newVal)
      }
    })
```

This looks good for me. Am I missing some behavior here?

I changed value prop be actually value, but not visible selectOption. Idk why it was wrong.

VaSelect.vue
```ts
    const selectContentPropsComputed = computed(() => ({
      ...pick(props, ['placeholder', 'autocomplete', 'multiple', 'disabled']),
      tabindex: tabIndexComputed.value,
      value: valueComputed.value, // <--
      valueString: valueComputedString.value,
      hiddenSelectedOptionsAmount: hiddenSelectedOptionsAmount.value,
      isAllOptionsShown: isAllOptionsShown.value,
      dropdownContentShown: showDropdownContent.value,
      focused: isInputFocused.value,
      getText,
    }))
```
![firefox_KZXCMLrqlQ](https://user-images.githubusercontent.com/23530004/216192409-836d2fff-7a2c-4899-a503-e6d9209f9d50.gif)

The last part to make is ask Select to clear autoCompleteInput text. With code above you can remove `dropdownContentShown` and `valueString` and pass instead `searchText`. Which will be cleared in VaSelect component after option is selected. 

I would like to see components where most of state is handled in that component and children components just used to show this state.

Reply:
Also, highlight color sucks. Should be primary with some opacity I guess.